third_party/ibis/ibis_oracle/registry.py

```diff
@@ -389,6 +389,11 @@ def _day_of_week_name(t, op):
 def _random(t, op):
     return sa.sql.literal_column("DBMS_RANDOM.VALUE")
 
+def _extract_epoch(t, op):
+    sub = operator.sub(sa.cast(t.translate(op.arg), sa.DATE), sa.sql.literal_column("DATE '1970-01-01'")).self_group()
+    mult = sa.cast(operator.mul(sub, sa.sql.literal_column("86400")), sa.NUMERIC)
```

Comment:
You might feel that this doesn't matter but I'll flag it anyway.

Using `sa.NUMERIC` here gives a final SQL expression of:

```
CAST((CAST(t0.cast_timestamp__col_tstz AS DATE) - DATE '1970-01-01') * 86400 AS NUMERIC)
```

`NUMERIC` works because it is an ANSI data type name. Oracle developers would expect to see `NUMBER` in their expressions.

```
mult = sa.cast(operator.mul(sub, sa.sql.literal_column("86400")), sa.dialects.oracle.NUMBER)
```

Giving:

```
CAST((CAST(t0.cast_timestamp__col_tstz AS DATE) - DATE '1970-01-01') * 86400 AS NUMBER)
```

Having typed that out I feel it's not a big deal. But I'll leave it here anyway for you to read and ignore :-)